cuda/matrix/sparsity_csr_kernels.cu

```diff
@@ -113,19 +113,30 @@ void classical_spmv(syn::value_list<int, subwarp_size>,
         c->get_values(),
         std::array<acc::size_type, 1>{
             {static_cast<acc::size_type>(c->get_stride())}});
-
+    if (c->get_size()[0] == 0 || c->get_size()[1] == 0) {
+        // empty output: nothing to do
+        return;
+    }
```

Comment:
Is this necessary with the guards below?

Reply:
right, they are the same thing. I will still keep the check in classical csr spmv because the checks are in different function and it's more complex